features/admin.applications.v1/utils/application-management-utils.ts

```diff
@@ -107,17 +107,20 @@ export class ApplicationManagementUtils {
     public static isApplicationOutdated(applicationVersion: string, grantTypes?: string[]): boolean {
 
         if (applicationVersion && grantTypes
-            && grantTypes.includes(ApplicationManagementConstants.CLIENT_CREDENTIALS_GRANT)) {
+            && grantTypes?.includes(ApplicationManagementConstants.CLIENT_CREDENTIALS_GRANT)) {
 
             const appVersionArray: number[] = applicationVersion?.match(/\d+/g).map(Number);
             const latestAppVersionArray: number[] = ApplicationManagementConstants
                 .LATEST_VERSION.match(/\d+/g).map(Number);
 
-            if (appVersionArray[0] < latestAppVersionArray[0]) {
+            if (appVersionArray.length > 0 && latestAppVersionArray.length > 0 &&
+                appVersionArray[0] < latestAppVersionArray[0]) {
                 return true;
-            } else if (appVersionArray[1] < latestAppVersionArray[1]) {
+            } else if (appVersionArray.length > 1 && latestAppVersionArray.length > 1 &&
+                appVersionArray[1] < latestAppVersionArray[1]) {
                 return true;
-            } else if (appVersionArray[2] < latestAppVersionArray[2]) {
+            } else if (appVersionArray.length > 2 && latestAppVersionArray.length > 2 &&
+                appVersionArray[2] < latestAppVersionArray[2]) {
```

Comment:
appVersionArray and latestAppVersionArray always has a length of 3.

Reply:
Fixed with 90b52b90446e6561b341dfe3e60fa0c16033a3cc